lib/beautiful/xresources.lua

```diff
@@ -70,13 +70,21 @@ local function get_screen(s)
     return s and screen[s]
 end
 
---- Get global or per-screen DPI value falling back to xrdb.
--- @tparam[opt] integer|screen s The screen.
+--- Get global or per-screen DPI value, falling back to xrdb.
+-- @screen[opt] s The screen.
 -- @treturn number DPI value.
 function xresources.get_dpi(s)
-    s = get_screen(s)
-    if dpi_per_screen[s] then
-        return dpi_per_screen[s]
+    if s then
+        s = get_screen(s)
+        if dpi_per_screen[s] then
+            return dpi_per_screen[s]
+        end
+        local width = 0
+        for _,v in pairs(s.outputs) do
+            width = width + v.mm_width
+        end
+        local inch = width / 25.4
+        return math.floor(s.workarea.width / inch)
```

Comment:
Well, a inch is 2.54cm, therefore the mm from `mm_width` get divided by `25.4` and you get inches.
I thought the variable name is good documentation already, but apparently it needs a comment like `25.4mm = 1inch`.